src/huggingface_hub/file_download.py

```diff
@@ -1188,3 +1188,41 @@ def _resumable_file_manager() -> "io.BufferedWriter":
         pass
 
     return pointer_path
+
+
+def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None, repo_type=None):
```

Comment:
```suggestion
def try_to_load_from_cache(
    repo_id: str,
    filename: str,
    cache_dir: Union[str, Path, None] = None,
    revision: Optional[str] = None,
    repo_type: Optional[str] = None,
) -> Optional[str]:
```

I would add type hints here.
Also make `cache_dir` optional and in such case default to `HUGGINGFACE_HUB_CACHE`

```py
    if cache_dir is None:
        cache_dir = HUGGINGFACE_HUB_CACHE
    if isinstance(cache_dir, Path):
        cache_dir = str(cache_dir)
```